azure-kusto-data/azure/kusto/data/aio/client.py

```diff
@@ -20,10 +20,18 @@
 @documented_by(KustoClientSync)
 class KustoClient(_KustoClientBase):
     @documented_by(KustoClientSync.__init__)
-    def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str]):
-        super().__init__(kcsb)
+    def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str], proxies: Optional[Dict[str, str]] = None):
+        super().__init__(kcsb, proxies)
+
+        if proxies is not None:
+            if len(proxies) > 1:
+                raise KustoClientError("Only one proxy is supported in async client")
```

Comment:
Why are you enforcing this?
Either let the underlying clients decide what's right and wrong or ask for a single proxy an build the dictionary yourself.
I vote on the former

Reply:
The thing is `requests` and almost everything else accepts a dictionary, but `aiohttp` accepts a single string.
The two options are either accept a dictionary for the async client, and pass it on to everything else, and for aiohttp directly convert it to string (that's what I did here, and that's why there is this check and the next comment).

The alternative is to accept a string for the async client, convert it to a dictionary and pass it forward.
I chose the first one because it keeps the interface the same between the sync/async clients.

So do you understand now? WDYT?

Reply:
I think the idea is that when your are using aiohttp you are already bound to a specific address which you need to select from the dictionary (or None if it is not there).
Does it make sense?

Reply:
Sorry I don't understand

Reply:
Solved offline